src/overlays/actors/ovl_Mir_Ray/z_mir_ray.c

```diff
@@ -30,25 +42,535 @@ const ActorInit Mir_Ray_InitVars = {
+};
+
+static ColliderJntSphItemInit sJntSphItemsInit[1] = {
+    {
+        { 0x00, { 0x00200000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x01, 0x00, 0x00 },
+        { 0, { { 0, 0, 0 }, 50 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x09, 0x00, 0x00, 0x00, COLSHAPE_JNTSPH },
+    1,
+    sJntSphItemsInit,
+};
+
+static MirRayDataEntry sMirRayData[] = {
+    { { -1160, 686, -880 }, { -920, 480, -889 }, 30, 50, 1.0f, 50, 150, 0.8f, 255, 255, 255, 0x02 },
+    { { -1856, 1092, -190 }, { -1703, 841, -186 }, 30, 70, 0.88f, 54, 150, 0.8f, 255, 255, 255, 0x02 },
+    { { 1367, 738, -860 }, { 1091, 476, -860 }, 30, 85, 0.0f, 0, 150, 0.8f, 255, 255, 255, 0x00 },
+    { { 2200, 1103, -220 }, { 2040, 843, -220 }, 30, 60, 0.0f, 0, 150, 0.8f, 255, 255, 255, 0x01 },
+    { { -560, 2169, -310 }, { -560, 1743, -310 }, 30, 70, 0.0f, 0, 150, 0.8f, 255, 255, 255, 0x00 },
+    { { 60, 1802, -1090 }, { 60, 973, -1090 }, 30, 70, 0.0f, 0, 150, 0.9f, 255, 255, 255, 0x0D },
+    { { 1140, 480, -860 }, { 1140, 480, -860 }, 30, 30, 1.0f, 10, 100, 0.9f, 255, 255, 255, 0x0E },
+    { { -560, 1743, -310 }, { -560, 1743, -310 }, 30, 30, 0.0f, 0, 100, 0.94f, 255, 255, 255, 0x0C },
+    { { 60, 1743, -310 }, { 60, 1743, -310 }, 30, 30, 0.0f, 0, 100, 0.94f, 255, 255, 255, 0x0C },
+    { { -1174, 448, 1194 }, { -1174, 148, 1194 }, 50, 100, 1.0f, 50, 150, 0.8f, 255, 255, 255, 0x03 }
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 0, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+extern Gfx D_060000B0[];
+extern Gfx D_06000C50[];
+
+void MirRay_SetupCollider(MirRay* this) {
+    Vec3f temp_vec;
+    MirRayDataEntry* temp_DataEntry;
+
+    temp_DataEntry = &sMirRayData[this->actor.params];
+    temp_vec.x = (this->poolPt.x - this->sourcePt.x) * temp_DataEntry->unk_10;
+    temp_vec.y = (this->poolPt.y - this->sourcePt.y) * temp_DataEntry->unk_10;
+    temp_vec.z = (this->poolPt.z - this->sourcePt.z) * temp_DataEntry->unk_10;
+    this->colliderSph.list->dim.worldSphere.center.x = temp_vec.x + this->sourcePt.x;
+    this->colliderSph.list->dim.worldSphere.center.y = temp_vec.y + this->sourcePt.y;
+    this->colliderSph.list->dim.worldSphere.center.z = temp_vec.z + this->sourcePt.z;
+    this->colliderSph.list->dim.worldSphere.radius = temp_DataEntry->unk_14 * this->colliderSph.list->dim.scale;
+}
+
+// Set up a light point between source point and reflection point. Reflection point is the pool point (for windows) or
+// at the player position (for mirrors) void MirRay_SetupLights(MirRay* this, GlobalContext* globalCtx) {
+void MirRay_MakeShieldLight(MirRay* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    MirRayDataEntry* temp_DataEntry = &sMirRayData[this->actor.params];
```

Comment:
i would name this temp in a way that matches our style, `dataEntry` or something more speicifc if you want, but remove `temp_`